debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/AbstractLogMinerEventProcessor.java

```diff
@@ -557,10 +557,38 @@ protected void handleSchemaChange(LogMinerEventRow row) throws InterruptedExcept
             return;
         }
 
+        final Scn commitScn = offsetContext.getCommitScn();
+        if (commitScn != null && commitScn.compareTo(row.getScn()) >= 0) {
+            LOGGER.trace("DDL: SQL '{}' skipped with {} (SCN) <= {} (commit SCN)", row.getRedoSql(), row.getScn(), commitScn);
+            return;
+        }
+
         LOGGER.trace("DDL: '{}' {}", row.getRedoSql(), row);
         if (row.getTableName() != null) {
             counters.ddlCount++;
             final TableId tableId = row.getTableId();
+
+            final int activeTransactions = getTransactionCache().size();
+            if (activeTransactions <= 1) {
```

Comment:
@Naros This is a bit convoluted. Could the code be restructured in a form of
```
if (activeTransactions == 0) {
}
else if (activeTransactions == 1) {
}
if (advanceLowerScnBoundary) {
}
It would be more linear and easier to udnerstand.